app/models/api_image.rb

```diff
@@ -6,4 +6,8 @@ class ApiImage < ApplicationRecord
 
   has_one_attached :original
   has_one_attached :glitched
+
+  def attach_key(key)
```

Comment:
S3 needs the files to land with a unique name so we have used SecureRandom uuid to generate this and attach it to the object.